usecases/api_key_usecase.go

```diff
@@ -54,65 +54,51 @@ func (usecase *ApiKeyUseCase) ListApiKeys(ctx context.Context) ([]models.ApiKey,
 	return apiKeys, nil
 }
 
-func (usecase *ApiKeyUseCase) getApiKey(ctx context.Context, exec repositories.Executor, apiKeyId string) (models.ApiKey, error) {
-	apiKey, err := usecase.apiKeyRepository.GetApiKeyById(ctx, exec, apiKeyId)
-	if err != nil {
-		return models.ApiKey{}, err
+func (usecase *ApiKeyUseCase) CreateApiKey(ctx context.Context, input models.CreateApiKeyInput) (models.ApiKey, error) {
+	apiKeyId := uuid.NewString()
+	key := generateAPiKey()
+	hash := sha256.Sum256([]byte(key))
+	apiKey := models.ApiKey{
+		Id:             apiKeyId,
+		Description:    input.Description,
+		Hash:           hash[:],
+		Key:            key,
+		OrganizationId: input.OrganizationId,
+		Role:           input.Role,
 	}
-	if err := usecase.enforceSecurity.ReadApiKey(apiKey); err != nil {
+
+	if err := usecase.enforceSecurity.CreateApiKey(input.OrganizationId); err != nil {
 		return models.ApiKey{}, err
 	}
-	return apiKey, nil
-}
 
-func (usecase *ApiKeyUseCase) CreateApiKey(ctx context.Context, input models.CreateApiKeyInput) (models.CreatedApiKey, error) {
-	apiKey, err := executor_factory.TransactionReturnValue(
+	if input.Role != models.API_CLIENT {
+		return models.ApiKey{}, errors.Wrap(
+			models.BadParameterError,
+			fmt.Sprintf("role %s is not supported", input.Role),
+		)
+	}
+
+	err := usecase.apiKeyRepository.CreateApiKey(
```

Comment:
untangling a bit of code here - we actually never need to read back the value from the DB, so no need for a transaction either